### PR TITLE
Add JsonFormatter

### PR DESCRIPTION
**Pull request by difince**
JsonFormatter added. 
This way no matter if go-mod-core-contracts
is used to create the request or not - printing the data in json
format happens in one place (JsonFormattter)

device list - use FormatWriter interface

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: pkg/formatters/interface.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package formatters
+
+type FormatWriter interface {
+	Write(obj interface{}) error
+}
```
Comment (mkbhanda) [inline pkg/formatters/interface.go:18]: Looks like a warning asking for a new line at end of file

File: pkg/formatters/emptyformatter.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package formatters
+
+type EmptyFormatter struct{}
+
+func (f *EmptyFormatter) Write(obj interface{}) (err error) {
+	return nil
+}
```
Comment (mkbhanda) [inline pkg/formatters/emptyformatter.go:20]: Looks like a warning asking for a new line at end of file

File: cmd/device/list/list.go [hunk 3]
```diff
@@ -29,45 +27,41 @@ import (
...
+		return
 	}
-	return cmd
+
+	formatter := createFormatter()
+	err = formatter.Write(devices)
+	return
+}
+
+func createFormatter() formatters.FormatWriter {
```
Comment (mkbhanda) [inline cmd/device/list/list.go:62]: You have this method defined in multiple places, consider refactoring, taking in arguments such as, string, template, and supported methods. Possibly defined in the same file as you define the empty formatter and interface. And if you did that do you still need an interface?

File: pkg/formatters/emptyformatter.go (hunk 2 above)
Comment (difince) [reply pkg/formatters/emptyformatter.go:20]: I removed the empty interface with the next PR . 
I but I have added here new line

File: cmd/device/list/list.go (hunk 3 above)
Comment (difince) [reply cmd/device/list/list.go:62]: yes, I intentionally named all these functions  with the same name.  
Yes, I could create one function that accepts these arguments in a single place and use it elsewhere... but i was thinking that in the future these functions could deffer and I was not sure how to approach it now ,so I intended to address it in follow-up PR when the idea is clear to me. 
But ok, let do it now, and if the these function start to defer ... we will address them

**Pull request merged**